datafusion/common/src/hash_utils.rs

```diff
@@ -386,9 +386,6 @@ pub fn create_row_hashes<'a>(
     random_state: &RandomState,
     hashes_buffer: &'a mut Vec<u64>,
 ) -> Result<&'a mut Vec<u64>> {
-    for hash in hashes_buffer.iter_mut() {
```

Comment:
I think the reason this code sets the values to zero is so that some code paths can reuse the same buffer for the hashes -- otherwise there may be left over hashes from previous batches. 

Maybe we could make something more clever that remembered when the buffer was being reused

Reply:
NM -- Now I see that @Dandandan  and @yjshen  are saying in https://github.com/apache/datafusion/pull/12083/files#r1723796994 we should just delete (all) this code 💯